app/ui/src/main/java/com/fsck/k9/ui/settings/account/AccountSettingsActivity.kt

```diff
@@ -114,3 +157,69 @@ class AccountSettingsActivity : K9Activity(), OnPreferenceStartScreenCallback {
         }
     }
 }
+
+class AccountSelectionSpinner : Spinner {
+    var selection: Account
+        get() = selectedItem as Account
+        set(value) {
+            val adapter = adapter as AccountsAdapter
+            Spinner@setSelection(adapter.getPosition(value))
+        }
+
+    var title: CharSequence = ""
+        set(value) {
+            val adapter = adapter as AccountsAdapter
+            adapter.title = value
+            adapter.notifyDataSetChanged()
+        }
+
+    private lateinit var cachedBackground: Drawable
+
+    constructor(context: Context) : super(context)
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
+
+    init {
+        adapter = AccountsAdapter(context)
+        cachedBackground = background
+    }
+
+    public fun loadAccounts(accounts: List<Account>) {
+        val adapter = adapter as AccountsAdapter
+        adapter.clear()
+        adapter.addAll(accounts)
+
+        setEnabled(accounts.size > 1)
+        background = if (accounts.size > 1) cachedBackground else null
+    }
+
+    internal class AccountsAdapter(context: Context) : ArrayAdapter<Account>(context, 0)  {
+        var title: CharSequence = ""
+
+        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+            val account = getItem(position)
+
+            val view = convertView
+                ?: LayoutInflater.from(context).inflate(R.layout.account_spinner_item, parent, false)
+
+            return view.apply {
+                setPadding(0, paddingTop, paddingRight, paddingBottom)
```

Comment:
IIRC it was to avoid having to duplicate the layout. I've added another layout for the drop-down and removed the uses of `setPadding`
No idea why I did it for both the item and drop-down